Added Recipient parameter to Java files

emilytran99 opened the pull request:
Added code in Message.java, MessageServlet.java, Datastore.java to support a Recipient parameter

Commit:
added recipient field for messages to java files

Commit:
Merge branch 'master' of https://github.com/sp19-codeu-20/codeu-starter-project into EmilyDev

aeftenie commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -67,6 +68,8 @@ public void storeMessage(Message message) {
         String text = (String) entity.getProperty("text");
         long timestamp = (long) entity.getProperty("timestamp");
+        String recipient = (String) entity.getProperty("recipient");
+        Message message = new Message(id, user, text, timestamp, recipient);
```
The message variable declared here is being rewritten on the next line. Which of them should be kept?

emilytran99 replied:
The first one, with the recipient parameter - forgot to take that second one out.

emilytran99 replied:
How do i make that change?

aeftenie replied:
I don't personally now, but it must be possible. I suggest you try searching it online or in the shared resources on the codeu website. Lauren may also know given that she has more experience with GitHub. But if you get stuck reach out to me again and I'll find some time to investigate it.

emilytran99 replied:
Figured it out!

aeftenie commented on src/main/java/com/google/codeu/data/Message.java:
```diff
@@ -41,6 +42,18 @@ public Message(UUID id, String user, String text, long timestamp) {
   }
 
+  public Message(String user, String text, String recipient) {
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
```
This constructor doesn't seem to be used. Is it needed?

emilytran99 replied:
I think the guide said that we were just adding in the constructor but not using it yet, to prep all the Java files for working with the recipient parameter.

aeftenie replied:
I'm also new to the familiarization project so correct me if I'm wrong, but I don't see where the guide says to add the constructors. To me, it seems like it asks to add the recipient parameter to the existent constructors.

aeftenie replied:
https://sites.google.com/codeustudents.com/spring-2019/weeks-1-2-familiarization-projects/direct-messages?authuser=1#h.p_wF3N6-wM4b6g

emilytran99 replied:
I read over the guide again and you're right - I had thought it meant to add another version of the constructor but I've updated it so that I just added the recipient parameter to the existing constructors.

aeftenie commented on src/main/java/com/google/codeu/data/Message.java:
```diff
@@ -56,4 +69,8 @@ public String getText() {
+
+  public String getRecipient(){
+  return recipient;
+}
```
The alignment of the last 2 lines is off a little.

emilytran99 replied:
Oops - I think Github is saying that since I'm the author of this pull request I can't make the change.

aeftenie commented on src/main/java/com/google/codeu/servlets/MessageServlet.java:
```diff
@@ -77,10 +77,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    String recipient = request.getParameter("recipient");
```
I am not familiar with the Serverlet so this may be a silly question, my apologies. How does this parameter gets in the request?

emilytran99 replied:
From my understanding I think "recipient" is the key and whatever the actual recipient is would be the value, and getParameter would get it for that specific entry? The guide said this shouldn't actually do anything yet.

aeftenie reviewed (changes requested):
Overall looks great! I only have a few comments to clarify some things.

Also, please rename the PR (pull request) to give it a meaningful title so that someone who only reads the title has an idea of what has changed without having to look at it.

Awesome job on keeping the PR short to help the review go faster!

emilytran99 committed:
Took out line with redundant Message constructor

had forgotten to delete the old version without recipient parameter

emilytran99 committed:
Updated indentation, removed extraneous constructors

Corrected indentation for getRecipient function and removed extraneous constructors

aeftenie commented:
LTGM! Can you please rebase your changes so that the checks can find the .stickler..yml file? That should get the to pass.
